core/src/mender-api.c

```diff
@@ -266,12 +328,42 @@ mender_api_check_for_deployment(mender_api_deployment_data_t *deployment) {
              mender_api_config.device_type);
 
     /* Perform HTTP request */
-    if (MENDER_OK
-        != (ret = mender_http_perform(mender_api_jwt, path, MENDER_HTTP_GET, NULL, NULL, &mender_api_http_text_callback, (void *)&response, &status))) {
+    if (MENDER_OK != (ret = mender_http_perform(mender_api_jwt, path, MENDER_HTTP_GET, NULL, NULL, &mender_api_http_text_callback, (void *)response, status))) {
         mender_log_error("Unable to perform HTTP request");
         goto END;
     }
 
+END:
+
+    /* Release memory */
+    if (NULL != path) {
+        free(path);
+    }
+
+    return ret;
+}
+
+mender_err_t
+mender_api_check_for_deployment(mender_api_deployment_data_t *deployment) {
+
+    assert(NULL != deployment);
+    mender_err_t ret      = MENDER_FAIL;
+    char        *response = NULL;
+    int          status   = 0;
+
+    if (MENDER_FAIL == (ret = api_check_for_deployment_v2(&status, (void *)&response))) {
+        goto END;
+    }
+
+    if (404 == status) {
```

Comment:
So `404` still means `MENDER_OK` above? I think this is a worth a comment, otherwise the code looks like it will never reach this.